src/libs/ajax/loaded-data/useLoadedData.ts

```diff
@@ -50,11 +64,15 @@ export const useLoadedData = <T>(): UseLoadedDataResult<T> => {
       const error = isFetchResponse(err) ?
         Error(await err.text()) :
         err
-      setLoadedData({
+      const errorResult: ErrorState<T, unknown> = {
         status: 'Error',
         state: previousState,
         error
-      })
+      }
+      setLoadedData(errorResult)
```

Comment:
It seems odd to me that the error result is now put in two different places, so client code has two different ways to be notified when an error occurs. Is there a need for both code paths? I don't see any use of this code path in this PR, and the linked ticket (https://broadworkbench.atlassian.net/browse/IA-3762) doesn't mention anything about adding this API.

My usual preference is to only add code when it's needed, to avoid keeping around unused code.

Reply:
the need came from nearby developer activity during this work... as the example states, onError is to accommodate patterns for side-effects.  A general desire is to bias towards allowing side-effects to be handled in layers outside the visual component... which this allows for by letting a consuming hook of the useInterestingData or useInterestingState pattern handle this side-effect and let the visual component just respond to the at-rest state object of { status: "Error", ... }

Reply:
I updated the PR comments to highlight the source of the need... and will make sure it gets into the merged commit description.

Reply:
OK, although I'd prefer to see a feature added only as its use is added. Otherwise it's hard to tell what code is being used or not in a code base. If I were to PR onto the code as-is, I might remove this feature, since it has no uses.